admin/modules/config/banning.php

```diff
@@ -35,6 +35,29 @@
 		$errors[] = $lang->error_filter_already_banned;
 	}
 
+	if($mybb->input['type'] == 1)
+	{
+		$ip_address = $db->escape_string($mybb->input['filter']);
+		$subnet_mask = "0";
+		if(strpos($ip_address, "*") !== false)
+		{
+			$ip_address = str_replace("*", "0", $ip_address);
+		}
+		else if(strpos($ip_address, "/") !== false)
+		{
+			list($ip_address, $subnet_mask) = explode("/", $ip_address);
+		}
+
+		$subnet_mask = intval($subnet_mask);
```

Comment:
This conversion obscures any errors that might be present in `$subnet_mask`; that is, the case in which the admin enters a value other than an integer for the subnet. Suggestion: replace the cast with an integer test, and append to `$errors` if the test fails. An appropriate test can be performed using the `ctype_digit()` function.

Reply:
Oh, good catch, thanks :)
Added the type check into the already existing validity check to assure that only one warning will be shown, if both IP address and subnet mask are invalid and to keep it compact.

Reply:
That works well too.